torch/lib/c10d/Utils.hpp

```diff
@@ -3,18 +3,124 @@
 #include <sys/socket.h>
 #include <sys/types.h>
 
-#include <cstdlib>
+#include <chrono>
 #include <cstdint>
+#include <cstdlib>
 #include <functional>
 #include <limits>
 #include <string>
 #include <system_error>
 #include <tuple>
 #include <vector>
-#include <chrono>
+
+#include <ATen/ATen.h>
+
+#include "Types.hpp"
+
+inline void hash_combine(std::size_t& seed) {}
+
+template <typename T, typename... Rest>
+inline void hash_combine(std::size_t& seed, const T& v, Rest... rest) {
+  std::hash<T> hasher;
+  seed ^= hasher(v) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+  hash_combine(seed, rest...);
+}
```

Comment:
This should probably include some reference to boost (possibly with a license text)

Reply:
Look into `torch/csrc/utils/hash.h`

Reply:
Nice, I'll just add that to the include path here. It's hacky, but I don't want to copy it.

@ezyang Do you want to move this file to aten?